chain-signatures/node/src/protocol/triple.rs

```diff
@@ -290,16 +284,6 @@ impl TripleManager {
         timeout: u64,
     ) -> Result<(), InitializationError> {
         let id = rand::random();
-
-        // Check if the `id` is already in the system. Error out and have the next cycle try again.
```

Comment:
This call is redundant since we check our in-memory and Redis storage for an id randomly created one line above. This check adds an unnecessary call to Redis (or 2, if we want to check "used triples").

Reply:
this check is not redundant because creating an id that collides with one of the ids already in our network is unlikely but still a possibility. So on the chance that we do, let's error out first and have it try again later. Otherwise, this id will actually make it all the way to triple generation phase, complete the triple generation, and then try to be inserted into storage which it will fail at that stage, taking up all the time with this process. It's better to check this first then to have this case

Reply:
What is the probability of that? My guess is ~ a chance to guess someone's private key. But calls to Redis are very real and happen often.

Reply:
It's very very unlikely it happens, but if it does, then it could break some things we did not intend for it to. Which is what I am trying to avoid. I would rather have us directly call redis in this case because the cost is not that bad. This is per triple generation anyways, so it doesn't matter. The biggest cost there is already the triple generation process